app/modules/wallet/wallet.controller.ts

```diff
@@ -0,0 +1,55 @@
+import * as restify from 'restify';
+import { InternalServerError, NotFoundError } from 'restify-errors';
+import IController from '../../interfaces/utils/IController';
+import IRequest from '../../interfaces/utils/IServer';
+import Wallet from '../../models/wallet.model';
+import { IWallet } from '../../interfaces/models';
+
+export default class WalletsController implements IController {
+  public async post(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletObj = req.body;
+      const wallet = <IWallet> await Wallet.create({
+        client_id: walletObj.client_id,
+        user_id: walletObj.user_id,
+        name: walletObj.name
+      });
+
+      return res.send({
+        wallet: wallet
+      });
+    } catch (e) {
+      console.log(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async get(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const wallet = await Wallet.findOne({ where: { id: req.params.wallet_id } });
+      if (!wallet) {
+        return res.send(new restify.NotFoundError('Wallet not found for given id'));
+      }
+      return res.send(wallet);
+    } catch (e) {
+      console.log(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async getAll(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const wallets = await Wallet.findAll({});
+      if (wallets.length === 0) {
```

Comment:
No need to check if its empty. Just send the `wallets` if nothing is found. It'll be empty which is not an error